stan/math/rev/mat/fun/cholesky_decompose.hpp

```diff
@@ -241,9 +384,19 @@ inline Eigen::Matrix<var, -1, -1> cholesky_decompose(
   check_symmetric("cholesky_decompose", "A", A);
```

Comment:
More on the debate on checks. For check_square I think leaving this here is fine, since performance wise this does the same thing (access to the the rows and cols members).

check_symmetric would be more performant if we did it with OpenCL, but that means that we would call check_symmetric at line 390 and before 393 but on L_A not A, so an unnecessary value_of_rec call happens if its actually not symmetric.

Perforamance wise this is nothing to worry about so I am fine leaving this as is for the sake of cleaner code.

Reply:
Yeah, the call to `value_of_rec` is also v expensive. I'll look at this later to see if there is a more clever way to do it but I'm fine with the CPU version

Reply:
Also a good example of where Seans previous PR that saves flags for when these checks pass on the data would solve this without having to do any gpu stuff